vllm/v1/engine/llm_engine.py

```diff
@@ -145,6 +183,10 @@ def add_request(
 
     def step(self) -> List[RequestOutput]:
 
+        if self.should_execute_dummy_batch:
+            self.should_execute_dummy_batch = False
+            # TODO: execute a dummy batch to sync across ranks
```

Comment:
Looks like this is not the right place for this logic? This should be in the EngineCore's busy loop I feel.

Reply:
I'm not familiar with the engine part, can you show me where i should put it?

Reply:
Should be in this loop I guess

https://github.com/vllm-project/vllm/blob/main/vllm/v1/engine/core.py#L305

Reply:
I added the code in [267cd82](https://github.com/vllm-project/vllm/pull/13591/commits/267cd82b6cdf6720cb8b1c78288a23868fc8b214), at least vllm's internal port usage will not conflict with the dp master ports.